source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -107,6 +107,20 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config& config, Http::HeaderMa
 
     new_zipkin_span =
         tracer.startSpan(config, request_headers.Host()->value().c_str(), start_time, context);
+
+  } else if (request_headers.XB3TraceId() && request_headers.XB3SpanId()) {
+    uint64_t parentId(0);
+    if (request_headers.XB3ParentSpanId()) {
+      parentId = std::stoull(request_headers.XB3ParentSpanId()->value().c_str(), nullptr, 16);
```

Comment:
Please use https://github.com/envoyproxy/envoy/blob/master/source/common/common/utility.h#L92 for string -> int conversions and handle error (also please add tests for that).